packages/pigeon/lib/dart_generator.dart

```diff
@@ -47,27 +47,28 @@ void _writeHostApi(DartOptions opt, Indent indent, Api api) {
           indent.writeln(encodedDeclaration);
         }
         final String channelName = makeChannelName(api, func);
-        indent.writeln('const BasicMessageChannel<Object$nullTag> channel =');
+        indent.writeln(
+            'const BasicMessageChannel<Object$nullTag> channel = BasicMessageChannel<Object$nullTag>(');
         indent.nest(2, () {
           indent.writeln(
-            'BasicMessageChannel<Object$nullTag>(\'$channelName\', StandardMessageCodec());',
+            '\'$channelName\', StandardMessageCodec());',
           );
         });
         final String returnStatement = func.returnType == 'void'
             ? '// noop'
             : 'return ${func.returnType}.decode(replyMap[\'${Keys.result}\']$unwrapOperator);';
-        indent.format(
-            '''final Map<Object$nullTag, Object$nullTag>$nullTag replyMap = await channel.send($sendArgument) as Map<Object$nullTag, Object$nullTag>$nullTag;
+        indent.format('''
+final Map<Object$nullTag, Object$nullTag>$nullTag replyMap =\n\t\tawait channel.send($sendArgument) as Map<Object$nullTag, Object$nullTag>$nullTag;
 if (replyMap == null) {
 \tthrow PlatformException(
 \t\tcode: 'channel-error',
 \t\tmessage: 'Unable to establish connection on channel.',
 \t\tdetails: null,
 \t);
 } else if (replyMap['error'] != null) {
-\tfinal Map<Object$nullTag, Object$nullTag> error = replyMap['${Keys.error}'] as Map<Object$nullTag, Object$nullTag>;
+\tfinal Map<Object$nullTag, Object$nullTag> error =\n\t\t\treplyMap['${Keys.error}']! as Map<Object$nullTag, Object$nullTag>;
 \tthrow PlatformException(
-\t\tcode: error['${Keys.errorCode}'] as String,
+\t\tcode: error['${Keys.errorCode}'] as String? ?? 'unknown',
```

Comment:
Sorry, this might conflict with this PR: https://github.com/flutter/flutter/issues/78336  I had to try to get that landed to address some customer's issues.

I think I like this solution better than the one I did there.  If you end up with a merge conflict, feel free to do this instead.